src/librustc/hir/lowering.rs

```diff
@@ -3143,7 +3143,24 @@ impl<'a> LoweringContext<'a> {
                 )
             }),
             ExprKind::Catch(ref body) => {
-                self.with_catch_scope(body.id, |this| hir::ExprBlock(this.lower_block(body, true)))
+                self.with_catch_scope(body.id, |this| {
+                    let unstable_span =
+                        this.allow_internal_unstable(CompilerDesugaringKind::Catch, body.span);
+                    let mut block = this.lower_block(body, true).into_inner();
+                    let tail = block.expr.take().map_or_else(
+                        || hir::Expr {
+                            id: block.id, // FIXME!
+                            span: block.span, // FIXME?
```

Comment:
this would be where we report errors regarding this `()` expression; the block seems ok, but we might want to .. just use the closing brace `}`?

Reply:
we probably also want to give it the "desugaring" span